src/batch/src/executor/insert.rs

```diff
@@ -110,15 +117,26 @@ impl InsertExecutor {
             let cap = chunk.capacity();
             let (mut columns, vis) = chunk.into_parts();
 
-            // No need to check for duplicate columns. This is already validated in binder.
-            if !&self.column_indices.is_sorted() {
-                let mut ordered_cols = columns.clone();
-                for (i, idx) in self.column_indices.iter().enumerate() {
-                    ordered_cols[*idx] = columns[i].clone()
-                }
-                columns = ordered_cols
+            let dummy_chunk = DataChunk::new_dummy(cap);
+
+            let mut ordered_columns = self
+                .column_indices
+                .iter()
+                .enumerate()
+                .map(|(i, idx)| (*idx, columns[i].clone()))
+                .collect_vec();
+
+            for (idx, expr) in &self.sorted_default_columns {
+                let column = Column::new(expr.eval(&dummy_chunk).await?);
+                ordered_columns.push((*idx, column));
```

Comment:
I guess the performance won't be too bad if we simply use `insert` here, since there won't be too many columns. 😄

Reply:
Ah that's not mainly about performance but to resolve potential insert index out of range😁